python/wikibot/wikibot.py

```diff
@@ -188,19 +188,35 @@ def ever_had_prop(self, wd_item, prop):
         if prop in claims: return True
     return False
 
+  def ever_had(self, wd_item, prop, target):
+    # Up to 150 revisions covers the full history of 99% human items
+    revisions = wd_item.revisions(total=150, content=True)
+    for revision in revisions:
+      try:
+        revision_text = json.loads(revision.text)
+        claims = revision_text['claims']
+      except:
+        pass # unable to extract claims - move to next revision
+      else:
+        if prop in claims:
+          for claim in claims[prop]:
+            old_claim = pywikibot.Claim.fromJSON(self.repo, claim)
+            if old_claim.target_equals(target): return True
+    return False
+
   def log_status_skip(self, item, facts, error):
-    print("Skipping", str(item), " -- ", str(facts), error)
+    print("Skipping", str(item.id), " -- ", str(facts), error)
```

Comment:
The print function will implicitly convert all arguments to string so the explicit str() conversions are not needed. You are welcome to postpone fixing this to a later PR.

Reply:
Thanks. Will fix in a later PR